test/jdk/sun/tools/jhsdb/JShellHeapDumpTest.java

```diff
@@ -81,6 +81,7 @@ public static void launch(String expectedMessage, List<String> toolArgs)
             System.out.println("###### End of all output which took " + elapsedTime + "ms");
             output.shouldHaveExitValue(0);
         } catch (Exception ex) {
+            if (allowRetry) return false;
```

Comment:
The exception should be printed if we are going to retry, and also maybe print a message that mentions retrying similar to what we do in printStackTraces()L

122                     System.out.println("'JShellToolProvider' missing. Allow one retry.");